src/assessments/seo/singleH1Assessment.js

```diff
@@ -0,0 +1,162 @@
+let AssessmentResult = require( "../../values/AssessmentResult.js" );
+let Assessment = require( "../../assessment.js" );
+let merge = require( "lodash/merge" );
+let map = require( "lodash/map" );
+let Mark = require( "../../values/Mark.js" );
+let marker = require( "../../markers/addMark.js" );
+
+/**
+ * Assessment to check whether the body of the text contains H1s in the body (except at the very beginning,
+ * where they are acceptable).
+ */
+class singleH1Assessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			scores: {
+				textContainsH1: 1,
+			},
+		};
+
+		this.identifier = "singleH1Assessment";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Runs the h1 research and based on this returns an assessment result with a score.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling the research.
+	 * @param {Object} i18n The object used for translations
+	 *
+	 * @returns {AssessmentResult} The assessment result.
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._h1s = researcher.getResearch( "h1s" );
+
+		let assessmentResult = new AssessmentResult();
+
+		// Return the default assessment result if there are no H1s in the body.
+		if ( this._h1s.length === 0 ) {
+			return assessmentResult;
+		}
+
+		/*
+		 * Remove the first H1 from the array if that H1 is in the first position of the body.
+		 * The very beginning of the body is the only position where an H1 is deemed acceptable.
+		 */
+		if ( this.firstH1AtBeginning() ) {
+			this._h1s.shift();
+		}
+
+		assessmentResult.setScore( this.calculateScore() );
+		assessmentResult.setText( this.translateScore( i18n ) );
+		assessmentResult.setHasMarks( ( this.calculateScore() < 2 ) );
+
+		return assessmentResult;
+	}
+
+	/**
+	 * Checks whether an H1 is in the first position of the body.
+	 *
+	 * @returns {boolean} Returns true if there is an H1 in the first position of the body.
+	 */
+	firstH1AtBeginning() {
+		if( this._h1s[ 0 ].position === 0 ) {
+			return true;
+		}
+	}
+
+	/**
+	 * Checks whether there is a single H1 in the body.
+	 *
+	 * @returns {boolean} Returns true if there is exactly one H1 in the body.
+	 */
+	bodyContainsOneH1() {
+		if ( this._h1s.length === 1 ) {
+			return true;
+		}
+	}
+
+	/**
+	 * Checks whether there are multiple H1s in the body.
+	 *
+	 * @returns {boolean} Returns true if the number of H1s in the body exceeds 1.
+	 */
+	bodyContainsMultipleH1s() {
+		if ( this._h1s.length > 1 ) {
+			return true;
+		}
+	}
+
+	/**
+	 * Returns the score for the single H1 assessment.
+	 *
+	 * @returns {number} The calculated score.
```

Comment:
We’ve discussed this point. I’l leave it as it is now, because when there’s no change in the values, the assessment result will just remain the default result and doesn’t have to be overwritten.

Reply:
I agree with @andizer on this one. However, it seems like there's something 'wrong' from an architectural standpoint that would need fixing. So, for the time being, I'd suggest leaving this as-is.